### PR TITLE
:seedling: TK-1581 : Added Log Collector

### PR DESCRIPTION
**Pull request by pallav-trilio**

### CODE REVIEW

**Review by msurudwad — CHANGES_REQUESTED** [review]
1. Remove `Setup Pre-requisuites of TVK License Creation` and `Authenticate and Connect to GKE cluster` from `.github/workflows/plugin-packages.yml`
2. Remove `tests/commons` dir as it's not needed anymore
3. Remove `hack/run-integration-tests.sh` as it's not needed anymore
4. Remove `testLogCollector` job variable changes wherever you find it.
5. Update `Run Preflight Integration Tests` step of preflight job to run shell test-suite.

@madhurtrilio please remove `TRILIODATA` repo secret as it's not needed anymore

File: .github/workflows/plugin-packages.yml [hunk 1]
```diff
@@ -71,15 +71,13 @@ jobs:
...
-              - 'tests/log-collector/**'
               - 'tests/common/**'
             testPreflight:
               - 'tools/preflight/**'
               - 'tests/preflight/**'
             testLogCollector:
               - 'cmd/**'
               - 'tools/log-collector/**'
-              - 'tests/log-collector/**'
               - 'tests/common/**'
```
[the remark this answers is not in the document]
Comment (pallav-trilio) [reply .github/workflows/plugin-packages.yml]: resolved

File: .github/workflows/plugin-packages.yml [hunk 2]
```diff
@@ -71,15 +71,13 @@ jobs:
               - '.krew/tvk-log-collector.yaml'
               - 'tools/log-collector/**'
               - '.goreleaser.yml'
-              - 'tests/log-collector/**'
               - 'tests/common/**'
```
[the remark this answers is not in the document]
Comment (pallav-trilio) [reply .github/workflows/plugin-packages.yml]: resolved

File: .github/workflows/plugin-packages.yml [hunk 3]
```diff
@@ -71,14 +70,12 @@ jobs:
               - '.krew/tvk-log-collector.yaml'
               - 'tools/log-collector/**'
               - '.goreleaser.yml'
-              - 'tests/common/**'
             testPreflight:
               - 'tools/preflight/**'
               - 'tests/preflight/**'
             testLogCollector:
```
Comment (msurudwad) [inline .github/workflows/plugin-packages.yml:76]: remove this whole `testLogCollector` section and wherever it's used

File: .github/workflows/plugin-packages.yml [hunk 4]
```diff
@@ -102,7 +99,7 @@ jobs:
         go-version: 1.15
 
     - name: Authenticate and Connect to GKE cluster
-      if: (needs.pre-build.outputs.testLogCollector == 'true') || startsWith(github.ref, 'refs/tags/')
```
Comment (msurudwad) [inline .github/workflows/plugin-packages.yml:107]: Do not remove this line... instead replace `testLogCollector` with  `testPreflight` here

File: .github/workflows/plugin-packages.yml [hunk 5]
```diff
@@ -111,9 +108,7 @@ jobs:
 
     - name: Run Preflight Integration Tests
       if: (needs.pre-build.outputs.testPreflight == 'true') || startsWith(github.ref, 'refs/tags/')
-      run: |
-        GO111MODULE=off go get -u github.com/onsi/ginkgo/ginkgo
-        ginkgo -r -keepGoing tests/preflight/
+      run: make test-preflight-intregation
```
Comment (msurudwad) [inline .github/workflows/plugin-packages.yml:111]: typo *integration

**Review by msurudwad — CHANGES_REQUESTED** [review]
> Remove `tests/commons` dir as it's not needed anymore

I don't see changes for this review comments... are you using it anywhere?

`Setup Go` step in `build_test_preflight` job is not needed anymore… remove this one too

File: .github/workflows/plugin-packages.yml (hunk 3 above)
Comment (pallav-trilio) [reply .github/workflows/plugin-packages.yml:76]: resolved

File: .github/workflows/plugin-packages.yml (hunk 4 above)
Comment (pallav-trilio) [reply .github/workflows/plugin-packages.yml:107]: resolved

File: .github/workflows/plugin-packages.yml (hunk 5 above)
Comment (pallav-trilio) [reply .github/workflows/plugin-packages.yml:111]: resolved

**Review by msurudwad — APPROVED** [review]

**Comment by madhurtrilio** [comment]
removed TRILIODATA secret

File: cmd/main.go [hunk 6]
```diff
@@ -1,7 +1,7 @@
 package main
```
Comment (madhurtrilio) [inline cmd/main.go:1]: Shouldn't this be under the cmd/log-collector dir?

Comment (msurudwad) [reply cmd/main.go:1]: Yes.. we can keep it that way no issues with that. If followed that, dir structure should be like - 
- cmd
  - log-collector 
    - cmd/log-collector.go
    - main.go 

There's supposed to be a single log-collector executable in `cmd` so did not bother to create separate and nested dir structure like above.

One thing that should be followed is executable(main.go) and commands(log-collector.go) file should be kept in separate dir, in order to keep them in different go packages.  
[ref](https://github.com/spf13/cobra#getting-started).

Comment (madhurtrilio) [reply cmd/main.go:1]: @pallav-trilio I am ok with the above structure suggested by @msurudwad

Comment (msurudwad) [reply cmd/main.go:1]: @pallav-trilio also need to update `.goreleaser.yml` for updated `main.go` path. Make sure to run `make build-log-collector` before git push.

Comment (pallav-trilio) [reply cmd/main.go:1]: Alright

Comment (pallav-trilio) [reply cmd/main.go]: resolved

**Review by madhurtrilio — APPROVED** [review]

**Pull request merged**